doc/adr/020_PayPal_IPN_Queue.md

```diff
@@ -0,0 +1,53 @@
+# PayPal IPN Queue
+
+Date: 2021-06-14
+
+Deciders: Kai Nissen, Gabriel Birke, Corinna Hillebrandt, Abban Dunne, Conny Kawohl
+
+Technical Story: https://phabricator.wikimedia.org/T284893
+
+## Status
+
+Declined
+
+## Context and Problem Statement
+
+The PayPal IPNs on Fundraising Frontend started to fail after a deployment 
+and was only noticed some days after when the Project Manager needed to 
+export the data. Upon investigation, it was discovered that:
+
+* Error logging was inactive on the application. (Now fixed)
+* We can’t debug using the responses our system returned to PayPal as 
+  we don’t have access to the IPN log.
+
+This led to a situation where we couldn't get the information required 
+to debug the error. It was suggested we queue all incoming requests from 
+PayPal on our own system for processing by our system.
+
+## Decision Drivers
+
+* **Transparency**: If our system fails we would have a stored queue to 
+  use for debugging.
+* **Automation**: The IPNs wouldn't need to be fired again once an error 
+  becomes fixed as our system would resume processing the queue.
+
+## Pros and Cons
+
+* Good, because debugging would be easier.
+* Good, because our system would resume processing automatically.
+* Good, because PayPal stops sending all IPNs when it starts to receive 
+  html 500 errors.
+* Bad, another system to support.
```

Comment:
```suggestion
* Bad, another system to support.
* Bad, because we send back each IPN to PayPal to verify that it's genuine.
  It's unclear how PayPal would behave if there is a time delay or 
  a duplicate verification (when re-processing old items).
```